source/backend/api/Services/DocumentFileService.cs

```diff
@@ -89,333 +91,509 @@ public IList<T> GetFileDocuments<T>(FileType fileType, long fileId)
             }
         }
 
-        public async Task<DocumentUploadRelationshipResponse> UploadResearchDocumentAsync(long researchFileId, DocumentUploadRequest uploadRequest)
+        public async Task UploadAcquisitionDocument(long acquisitionFileId, DocumentUploadRequest uploadRequest)
         {
-            Logger.LogInformation("Uploading document for single research file");
-            User.ThrowIfNotAllAuthorized(Permissions.DocumentAdd, Permissions.ResearchFileEdit);
+            Logger.LogInformation("Uploading document for single acquisition file");
+            User.ThrowIfNotAllAuthorized(Permissions.DocumentAdd, Permissions.AcquisitionFileEdit);
+            uploadRequest.ThrowInvalidFileSize();
 
-            // Do not call Mayan if uploaded file is empty (zero-size)
-            ValidateZeroLengthFile(uploadRequest);
+            using var transaction = _documentQueueRepository.BeginTransaction();
 
-            DocumentUploadResponse uploadResult = await documentService.UploadDocumentAsync(uploadRequest);
+            // Step 1 - Save the pimsDocument
+            PimsDocument pimsDocument = CreatePimsDocument(uploadRequest);
+            _documentQueueRepository.SaveChanges();
 
-            DocumentUploadRelationshipResponse relationshipResponse = new DocumentUploadRelationshipResponse()
+            // Step 2 - Create the File Document
+            PimsAcquisitionFileDocument newAcquisitionDocument = new()
             {
-                UploadResponse = uploadResult,
+                AcquisitionFileId = acquisitionFileId,
+                DocumentId = pimsDocument.DocumentId,
             };
+            _acquisitionFileDocumentRepository.AddAcquisition(newAcquisitionDocument);
+            _documentQueueRepository.SaveChanges();
 
-            // Throw an error if Mayan returns a null document. This means it wasn't able to store it.
-            ValidateDocumentUploadResponse(uploadResult);
-
-            if (uploadResult.Document is not null && uploadResult.Document.Id != 0)
+            // Step 3 - Queue the Document for processing
+            PimsDocumentQueue queueDocument = new()
             {
-                // Create the pims document research file relationship
-                PimsResearchFileDocument newResearchFileDocument = new PimsResearchFileDocument()
-                {
-                    ResearchFileId = researchFileId,
-                    DocumentId = uploadResult.Document.Id,
-                };
-                newResearchFileDocument = researchFileDocumentRepository.AddResearch(newResearchFileDocument);
-                researchFileDocumentRepository.CommitTransaction();
-
-                relationshipResponse.DocumentRelationship = mapper.Map<DocumentRelationshipModel>(newResearchFileDocument);
-            }
+                DocumentId = pimsDocument.DocumentId,
+                AcquisitionFileDocumentId = newAcquisitionDocument.AcquisitionFileDocumentId,
```

Comment:
You can take this line out, as we are removing all file foreign key relationships from the document queue table. Same goes for each of the file types.

Reply:
removed